### PR TITLE
Makes diagonal wall containing turfs lit properly

### LINKED ISSUE #31065: Shuttle Corner Pieces Are Space-Lit
(WEB REPORT BY: primed REMOTE: 206.221.180.138:7777)
# Revision
f078d50999f549a053630cecb5a15f28c9522526

# Description
The corners of shuttles and escape pods are fully lit up, as if they were space. This looks incorrect, since all the walls and other materials around it are lit correctly.

### PR DESCRIPTION
**Pull request by ghost**
[bugfix][system]
Closes #31065
:cl:
 * bugfix: Diagonal shuttle walls are properly lit once again.

### CODE REVIEW

File: code/game/turfs/simulated/shuttle.dm [hunk 1]
```diff
@@ -76,6 +76,36 @@
...
+/obj/structure/shuttle/diag_wall/initialize()
+	var/turf/T = get_turf(src)
+	if(T)
+		T.dynamic_lighting = 1
+		if(SSlighting && SSlighting.initialized && !T.lighting_overlay)
+			new /atom/movable/lighting_overlay(T, TRUE)
+
+/obj/structure/shuttle/diag_wall/New()
+	..()
+	initialize()
```
Comment (DamianX) [inline code/game/turfs/simulated/shuttle.dm]: check `if(world.has_round_started())` before calling this or it will initialize twice at round-start

Comment (ghost) [reply code/game/turfs/simulated/shuttle.dm]: Done

File: code/game/turfs/simulated/shuttle.dm [hunk 2]
```diff
@@ -76,6 +76,38 @@
...
+/obj/structure/shuttle/diag_wall/initialize()
+	var/turf/T = get_turf(src)
+	if(T)
+		T.dynamic_lighting = 1
+
+/obj/structure/shuttle/diag_wall/New()
+	..()
+	if(world.has_round_started())
+		initialize()
+	var/turf/T = get_turf(src)
```
Comment (DamianX) [inline code/game/turfs/simulated/shuttle.dm]: what? Why do this here?

Comment (ghost) [reply code/game/turfs/simulated/shuttle.dm]: It only worked in initialisation, moved it back in with old checks anyways.

**Pull request merged**